src/components/Awareness.tsx

```diff
@@ -7,20 +7,31 @@ import { useAwareness } from '@/hooks'
 import { Collaboration } from '@/defaults'
 
 interface AwarenessProps extends PropsWithChildren {
-  id: string
+  name: string
+  visual?: boolean
 }
 
-export const Awareness = React.forwardRef(({ id, children }: AwarenessProps, ref) => {
-  const [focused, setIsFocused] = useAwareness(id)
+/**
+ * Wrapper around anything that should be able to indicate awareness locally
+ * as well as signal awareness to remote users. Children use the ref function
+ * to signal awareness using true/false.
+ *
+ * If prop `visual` is false, it only signals awareness
+ */
+export const Awareness = React.forwardRef(({ name, visual = true, children }: AwarenessProps, ref) => {
+  const [states, setIsFocused] = useAwareness(name)
   const awarenessRef = useRef(null)
 
   useImperativeHandle(ref, () => {
     return setIsFocused
   })
 
-  const color = Collaboration.colors[focused?.color || ''] || ''
+  // Currently we only show one user color as ring around remotely focused element
+  const state = states.length ? states[0] : null
+  const color = Collaboration.colors[state?.focus?.color || '']?.ring || ''
+
   return (
-    <div ref={awarenessRef} className={`rounded ${focused ? 'ring' : ''} ${color}`}>
+    <div ref={awarenessRef} className={`rounded ${visual && !!color ? 'ring' : ''} ${color}`}>
```

Comment:
Feels like this could be clearer. `state` isn't typed as possible undefined, yet we check for length.
Maybe use `cn()` instead of ternary?